sql/mysqld.cc

```diff
@@ -5033,6 +5033,21 @@ a file name for --log-bin-index option", opt_binlog_index_name);
   }
   plugins_are_initialized= TRUE;  /* Don't separate from init function */
 
+  LEX_CSTRING keyring_vault_name= {C_STRING_WITH_LEN("keyring_vault")};
+  LEX_CSTRING keyring_name= {C_STRING_WITH_LEN("keyring_file")};
+  if ((plugin_is_ready(keyring_vault_name, MYSQL_KEYRING_PLUGIN) ||
+       plugin_is_ready(keyring_name, MYSQL_KEYRING_PLUGIN)) &&
+      !pxc_encrypt_cluster_traffic)
```

Comment:
I would suggest to reorder the checks performed here. `plugin_is_ready()` locks `LOCK_plugin` and therefore is a costly operation. So, here is my suggestion.
```suggestion
  if (!pxc_encrypt_cluster_traffic &&
      (plugin_is_ready(keyring_vault_name, MYSQL_KEYRING_PLUGIN) ||
       plugin_is_ready(keyring_name, MYSQL_KEYRING_PLUGIN))
```